File: _BHCVRP_Python_Version/cujae_inf_citi_om/_vrp_data/Customer.py
from _vrp_data import Location

# Clase que modela los datos de un cliente en un VRP.

class Customer:
    
    def __init__(self, id_customer, request_customer, location_customer: Location):
        self._id_customer = id_customer
        self._request_customer = request_customer
        self._location_customer = location_customer

    def get_id_customer(self):
        return self._id_customer

    def set_id_customer(self, id_customer):
        self._id_customer = id_customer

    def get_request_customer(self):
        return self._request_customer

    def set_request_customer(self, request_customer):
        self._request_customer = request_customer

    def get_location_customer(self):
        return self._location_customer

    def set_location_customer(self, location_customer: Location):
        self._location_customer = location_customer

File: BHCVRP_Java_Version/src/cujae/inf/citi/om/data/Customer.java
package cujae.inf.citi.om.data;

/* Clase que modela los datos de un cliente en un VRP*/

public class Customer {
	
	protected int idCustomer;
	protected double requestCustomer;
	protected Location locationCustomer;
        protected TimeWindow timeWindow;

	public Customer() {
		super();
	}
	
	public Customer(int idCustomer, double requestCustomer,
			Location locationCustomer) {
		super();
		this.idCustomer = idCustomer;
		this.requestCustomer = requestCustomer;
		this.locationCustomer = locationCustomer;
	}
        
        public Customer(int idCustomer, double requestCustomer,
			Location locationCustomer, TimeWindow timeWindow) {
		super();
		this.idCustomer = idCustomer;
		this.requestCustomer = requestCustomer;
		this.locationCustomer = locationCustomer;
                this.timeWindow = timeWindow;
	}

	public int getIdCustomer() {
		return idCustomer;
	}

	public void setIdCustomer(int idCustomer) {
		this.idCustomer = idCustomer;
	}

	public double getRequestCustomer() {
		return requestCustomer;
	}

	public void setRequestCustomer(double requestCustomer) {
		this.requestCustomer = requestCustomer;
	}

	public Location getLocationCustomer() {
		return locationCustomer;
	}

	public void setLocationCustomer(Location locationCustomer) {
		this.locationCustomer = locationCustomer;
	}
        
        public TimeWindow getTimeWindow(){
            return timeWindow;
        }
        
        public void setTimeWindow(TimeWindow timeWindow){
            this.timeWindow = timeWindow;
        }
}
